connectors/sources/confluence.py

```diff
@@ -112,6 +145,36 @@ async def api_call(self, url):
                 )
                 await self._sleeps.sleep(RETRY_INTERVAL**retry_counter)
 
+    async def paginated_api_call(self, url_name, **url_kwargs):
+        """Make a paginated API call for Confluence objects using the passed url_name.
+        Args:
+            url_name (str): URL Name to identify the API endpoint to hit
+        Yields:
+            response: JSON response.
+        """
+        should_paginate = True
+        url = os.path.join(self.host_url, URLS[url_name].format(**url_kwargs))
+        while should_paginate:
```

Comment:
Do we even need `should_paginate` or could we just use `while True` and simply break, if we encounter a condition to break?

Reply:
Yes, we can use directly `while True` then `break` inplace of a variable like `should_paginate`. I kept it this way as it maintains the code readability otherwise the structure becomes like below:

```
while True:
   ...
   if links.get("next") is None:
      break    # to come out of async for loop
      break    # to come out of pagination (the while loop)
```

My only concern is two breaks one after the other is a bit confusing to understand.

Reply:
Couldn't you just return?

Reply:
Correct, thanks, Updated with commit id: [7880cf0](https://github.com/elastic/connectors-python/pull/604/commits/7880cf0cd4d5c1b14509a6b7659b7af65bc54f50)